app/src/main/java/org/oppia/android/app/devoptions/featureflags/FeatureFlagItemViewModel.kt

```diff
@@ -0,0 +1,79 @@
+package org.oppia.android.app.devoptions.featureflags
+
+import androidx.annotation.ColorInt
+import androidx.databinding.ObservableField
+import org.oppia.android.app.model.FeatureFlagId
+import org.oppia.android.app.model.SyncStatus
+import org.oppia.android.app.translation.AppLanguageResourceHandler
+import org.oppia.android.app.view.models.R
+import org.oppia.android.app.viewmodel.ObservableViewModel
+import org.oppia.android.util.locale.OppiaLocale
+import javax.inject.Inject
+
+/** [ViewModel] for displaying a feature flag for the recycler view in [FeatureFlagsFragment]. */
+class FeatureFlagItemViewModel @Inject constructor(
+  val featureFlagId: FeatureFlagId,
+  val currentValue: Boolean,
+  val syncStatus: SyncStatus,
+  private val machineLocale: OppiaLocale.MachineLocale,
+  private val resourceHandler: AppLanguageResourceHandler
+
+) : ObservableViewModel() {
+
+  val isChecked: ObservableField<Boolean> = ObservableField(currentValue)
+  val featureFlagDisplayName: ObservableField<String> =
+    ObservableField(getFeatureFlagDisplayName(featureFlagId))
+  val syncStatusDisplayText: ObservableField<String> =
+    ObservableField(getSyncStatusText())
+
+  var onToggleCallback: ((FeatureFlagId, Boolean) -> Unit)? = null
+
+  fun onUserToggle() {
```

Comment:
`onToggleFeatureFlag` or more descriptive names that specifies the action item (as suggested (https://github.com/oppia/oppia-android/pull/5861/files/027ba66557c4a435c730a129ecb4f0d6cdc97c3f#r2158892771) that was a sample and mentioned below to user appropriate variable names - please use suggestions as quick references and utilize according to the actual usages)
```suggestion
  fun onToggleFeatureFlag() {
```

Reply:
This applies for `onToggleCallback` as well.

Reply:
updated